mono/metadata/sre.c

```diff
@@ -2350,14 +2350,36 @@ mono_reflection_get_custom_attrs_blob_checked (MonoReflectionAssembly *assembly,
 static gboolean
 reflection_setup_internal_class (MonoReflectionTypeBuilderHandle ref_tb, MonoError *error)
 {
+	static GHashTable *seen = NULL;
```

Comment:
I think we can fix this by renaming reflection_setup_internal_class to reflection_setup_internal_class_internal while adding a new function named reflection_setup_internal_class.

We'd do something like this: ```
static gboolean
reflection_setup_internal_class (MonoReflectionTypeBuilderHandle ref_tb, MonoError *error)
{
   MonoPtrArray types;
   mono_ptr_array_init (types, `6, MONO_ROOT_SOURCE_REFLECTION, "sre setup internal class");
   gboolean res = reflection_setup_internal_class_internal (ref_tb, &types, error);
   mono_ptr_array_destroy (types);
   return res;
}
```

And in reflection_setup_internal_class_internal we'd search `types` for what we're init'ing and add to it if not found. The average depth will never be big enough for this to matter.